src/components/army.cairo

```diff
@@ -0,0 +1,58 @@
+use eternum::components::buildings::Buildings;
+
+//
+// ----------- Army
+// Armies can be attacked to any Entity if it has a Spawn function. Realms can spawn them and have ownership over them, however they can exist
+// outside of a Realm. They might spawn randomly on the Map.
+
+#[derive(Component)]
+struct Army {
+    owner_entity_id: u32, // Armies are Entities. We assign an Army to a Realm for example, to give ownership to the Realm.
+    light_cavalry_qty: u32,
+    light_cavalry_health: u32,
```

Comment:
good question, i think that if we need to create a struct, maybe it means that the abstraction is too high it should be its own component?

Reply:
How would that look? Can you expand on it a little please?

Reply:
for example you'd have the an Army component like this:
```
struct Army {
        army_type: u8,
        qty: u32,
        health: u8,
}
```
and an army would be spawned by creating an entity with keys (realm_id, army_id, army_type).

To go even further you could:
- add `Speed` component to the army
- use `Health` component instead of health field in `Army`
- add `Attack` component

Reply:
Whoa, that sounds like a great idea! 🧠 

I'll resolve this convo, but let's revisit this once we tackle armies and combat.